File: root/trunk/source/modules/strategy/src/test/sample_data/inputs/SendOther.java
import java.math.BigDecimal;

import org.marketcetera.event.AskEvent;
import org.marketcetera.event.TradeEvent;
import org.marketcetera.strategy.java.Strategy;

/* $License$ */

/**
 * Sample strategy to test the ability to send arbitrary data.
 *
 * @author <a href="mailto:colin@marketcetera.com">Colin DuPlantis</a>
 * @version $Id$
 * @since 2.0.0
 */
public class SendOther
        extends Strategy
{
    /* (non-Javadoc)
     * @see org.marketcetera.strategy.java.Strategy#onAsk(org.marketcetera.event.AskEvent)
     */
    @Override
    public void onAsk(AskEvent inAsk)
    {
        if(getProperty("sendNull") != null) {
            send(null);
        } else if(getProperty("sendString") != null) {
            send("test string");
        } else if(getProperty("sendTwo") != null) {
            send(BigDecimal.ONE);
            send(BigDecimal.TEN);
        }
    }
}

File: modules/strategy/src/test/sample_data/inputs/send_other.rb
include_class "org.marketcetera.strategy.ruby.Strategy"
include_class "org.marketcetera.trade.Equity"
include_class "java.math.BigDecimal"
include_class "java.lang.System"
include_class "java.lang.Boolean"

class SendOther < Strategy
    def on_ask(ask)
        if(get_property("sendNull") != nil)
            send nil
            return
        end
        if(get_property("sendString") != nil)
            send "test string"
            return
        end
        if(get_property("sendTwo") != nil)
            send BigDecimal::ONE
            send BigDecimal::TEN
        end
    end
end
